### PR TITLE
RB-347 separate delivery events

### PR DESCRIPTION
**Pull request by ziklodi**
we need to create two tables for callback events:
* delivery events: delivery | bounce
* engagement events: open | click | complaint

### CODE REVIEW

File: spec/sincerely/services/events/aws_ses_event_spec.rb [hunk 1]
```diff
@@ -48,10 +48,8 @@
         message_id: 'EXAMPLE7c191be45-e9aedb9a-02f9-4d12-a87d-dd0099a07f8a-000000',
         recipient: 'recipient@example.com',
         timestamp: Time.parse('2017-08-05T00:41:02.669Z'),
-        options: {
-          complaint_feedback_type: 'abuse',
-          user_agent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36' # rubocop:disable Layout/LineLength
-        }
+        feedback_type: 'abuse',
+        user_agent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36' # rubocop:disable Layout/LineLength
```
Comment (willianveiga) [inline spec/sincerely/services/events/aws_ses_event_spec.rb:52]: Probably you can avoid the `rubocop:disable` here with something like (not tested):
```suggestion
        user_agent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) ' \
                            'Chrome/60.0.3112.90 Safari/537.36'
```

File: spec/sincerely/services/events/aws_ses_event_spec.rb [hunk 2]
```diff
@@ -108,11 +104,9 @@
         message_id: 'EXAMPLE7c191be45-e9aedb9a-02f9-4d12-a87d-dd0099a07f8a-000000',
         recipient: 'recipient@example.com',
         timestamp: Time.parse('2017-08-09T23:51:25.570Z'),
-        options: {
-          ip_address: '192.0.2.1',
-          user_agent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36', # rubocop:disable Layout/LineLength
-          link: 'http://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-email-smtp.html'
-        }
+        ip_address: '192.0.2.1',
+        user_agent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36', # rubocop:disable Layout/LineLength
```
Comment (willianveiga) [inline spec/sincerely/services/events/aws_ses_event_spec.rb:108]: [Same](https://github.com/100Starlings/sincerely/pull/19/files#r1699870168) here.

File: spec/sincerely/services/process_delivery_event_spec.rb [hunk 3]
```diff
@@ -36,6 +36,10 @@
 
     context 'when bounce event' do
       let(:event_type) { 'bounce' }
+      let(:event) do
+        double(:event, event_type:, message_id:, recipient:, timestamp:, delivery_system:, options:, # rubocop:disable RSpec/VerifiedDoubles
```
Comment (willianveiga) [inline spec/sincerely/services/process_delivery_event_spec.rb:40]: Do we want to disable the `RSpec/VerifiedDoubles` cop globally? I think so...

Comment (ziklodi) [reply spec/sincerely/services/process_delivery_event_spec.rb:40]: > Do we want to disable the `RSpec/VerifiedDoubles` cop globally? I think so...

I think forcing `instance_double` is required, however in these case it is an exception as the event is a polymorphic object

**Review by willianveiga — APPROVED** [review]

**Pull request merged**